src/Modules/UngradedModules.jl

```diff
@@ -4606,6 +4655,37 @@ function free_resolution(M::SubQuo; ordering::ModuleOrdering = default_ordering(
   return FreeResolution(cc)
 end
 
+function free_resolution(M::SubQuo{T}) where {T<:RingElem}
+  # This generic code computes a free resolution in a lazy way.
+  # We start out with a presentation of M and implement 
+  # an iterative fill function to compute every higher term 
+  # on request.
+  R = base_ring(M)
+  p = presentation(M)
+  p.fill = function(C::Hecke.ChainComplex, k::Int)
+    # TODO: Use official getter and setter methods instead 
+    # of messing manually with the internals of the complex.
+    for i in first(range(C)):k-1
+      N = domain(map(C, i))
+
+      if iszero(N) # Fill up with zero maps
+        phi = hom(N, N, elem_type(N)[])
+        pushfirst!(C.maps, phi)
+        continue
+      end
+
+      K, inc = kernel(map(C, i))
+      nz = findall(x->!iszero(x), gens(K))
+      F = FreeMod(R, length(nz))
+      iszero(length(nz)) && set_attribute!(F, :name => "0")
+      phi = hom(F, C[i], iszero(length(nz)) ? elem_type(C[i])[] : inc.(gens(K)[nz]))
+      pushfirst!(C.maps, phi)
+    end
+    return first(C.maps)
+  end
```

Comment:
The method does **not** compute the resolution up to a user-specified upper bound, but in a lazy way, incorporated already in the chain complexes.

@fieker: Is this how you intended people to do it? 

I am a bit worried, since I am using the fields of the chain complex directly. But I didn't find any constructor that would accept the `fill` function as an argument. If you have methods for whatever I am doing here: Could you point them out to us?